### ISSUE TITLE
Copy button and Show more/Show less toggle prevent the line endings of code samples from being visible

### ISSUE THREAD

**Issue by mcorbin-ibm** [carbon-design-system/carbon#5689]
On IBM Developer, in code listings, the "Copy" button and "Show more"/"Show less" toggle prevent the line endings of code samples from being visible. The user is not able to see the full line of code unless they click the "Copy" button and paste the code elsewhere (note, text editor, etc).

Examples can be found throughout this tutorial:
https://developer.ibm.com/tutorials/monitor-spring-boot-microservices/

![image](https://user-images.githubusercontent.com/37812332/77462978-3f91b300-6ddb-11ea-8a03-8c6bc70559ff.png)

**Comment by tw15egan** [comment]
@carbon-design-system/design what do you recommend in this scenario?

**Comment by aagonzales** [comment]
I think there are two possibilities: 
1. Make the "show more/less" behavior optional so it can be turned off if not needed. The we'd have to increase the right padding so the text ended before the edge, preventing it from running under the copy button. We do add extra right padding to many content component so it wouldn't be outside the system to do. 
2. The other option would be to move the copy button to the bottom along with the show more/less function and just not allow text in that area. This would move the copy button when the snippet is expanded. 

![image](https://user-images.githubusercontent.com/11670886/77470632-ff343400-6dde-11ea-9745-4e2574d4f697.png)

**Comment by aagonzales** [comment]
Actually, since you can horizontally scroll you can see most of the content until you come to the end, so if we could somehow build padding into the end of the scrolling then it wouldn't have to be adjusted. Would that be possible? 

![Mar-24-2020 14-54-42](https://user-images.githubusercontent.com/11670886/77471093-b630af80-6ddf-11ea-92f0-759152d147eb.gif)

**Comment by mcorbin-ibm** [comment]
If I can weigh in, I would love for the Show More/Less button to not display if fewer than 3/4 lines of code.

Also, I would rather that the Show More/Less be lowered so that it did not obstruct any lines of code, even with the horizontal scrolling.

**Comment by tw15egan** [comment]
In React, we have it set up so that if the height is less than `255px` it will not show the `show more` button. If you're using Vanilla, you can just remove it for shorter code examples. 

What version of the components are you running?

**Comment by mcorbin-ibm** [comment]
I am told by Jamie Pickett (who cannot comment here):
"We are using 10.3.1. We are using Vanilla Javascript."

----
Also, please note that this is an issue for both Desktop & Mobile sites, not just mobile.

**Comment by tw15egan** [comment]
For vanilla, you can simply remove the `show more` button on smaller code snippets, like so

```html
<div class="bx--snippet bx--snippet--multi" 
  data-code-snippet>
  <div  class="bx--snippet-container"
    aria-label="Code Snippet Text">
    <pre>
       <code>Your Code Here</code>
    </pre>
  </div>
  <button data-copy-btn class="bx--copy-btn" type="button" tabindex="0">
    <span class="bx--assistive-text bx--copy-btn__feedback">Copied!</span>
    <svg focusable="false" preserveAspectRatio="xMidYMid meet" style="will-change: transform;" xmlns="http://www.w3.org/2000/svg" class="bx--snippet__icon" width="16" height="16" viewBox="0 0 16 16" aria-hidden="true"><path d="M14,5v9H5V5h9m0-1H5A1,1,0,0,0,4,5v9a1,1,0,0,0,1,1h9a1,1,0,0,0,1-1V5a1,1,0,0,0-1-1Z"></path><path d="M2,9H1V2A1,1,0,0,1,2,1H9V2H2Z"></path></svg>
  </button>
</div>
```

**Comment by mcorbin-ibm** [comment]
Yes, but will you make a change to handle adjusting these buttons for larger code snippets -- longer lines and larger number of lines of code -- so that they do not overlap/cover the code snippets?

**Comment by tw15egan** [comment]
@aagonzales do we want to add padding around these elements so that they do not overlap any text? 

I think unless we did something like your 2nd example, we would have to add quite a bit of padding to the bottom and it would seem a bit top-heavy.

**Comment by aagonzales** [comment]
No, I think the solution we have in the PR solves these problems without having to adjust and rearrange elements within the snippet. I have confirmed it with Jeannie, our CD. Even though the text are underneath the elements to start once the user scrolls or expands the window they can see what was once hidden. We've add in padding to ensure the the far right and bottom of the content is not hidden.

**Comment by tw15egan** [comment]
Fixed via #5704
